Classes/Emogrifier.php

```diff
@@ -277,14 +307,16 @@ function (array $m) {
             }
         }
 
-        // now iterate through the nodes that contained inline styles in the original HTML
-        foreach ($this->styleAttributesForNodes as $nodePath => $styleAttributesForNode) {
-            $node = $this->visitedNodes[$nodePath];
-            $currentStyleAttributes = $this->parseCssDeclarationBlock($node->getAttribute('style'));
-            $node->setAttribute(
-                'style',
-                $this->generateStyleStringFromDeclarationsArrays($currentStyleAttributes, $styleAttributesForNode)
-            );
+        // If enabled, iterate through the nodes that contained inline styles in the original HTML
+        if ($this->isInlineStylesParsingEnabled) {
+            foreach ($this->styleAttributesForNodes as $nodePath => $styleAttributesForNode) {
```

Comment:
While you're changing this part anyway, could you please move it to a separate private method?